File: easy/242-valid-anagram/242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) return false;
    const sCount = {}, tCount = {};
    for (let i = 0; i < s.length; i++) {
        sCount[s[i]] = ++sCount[s[i]] || 1
        tCount[t[i]] = ++tCount[t[i]] || 1
    }
    for (const key in sCount) {
        if (sCount[key] !== tCount[key]) return false;
    }
    return true;
};

File: easy/242-valid-anagram/242-valid-anagram.rb
# @param {String} s
# @param {String} t
# @return {Boolean}
def is_anagram(s, t)
    freq = {}
    s.each_char { |c| freq[c] = (freq[c] || 0) + 1}
    t.each_char do |c|
        return false if !freq[c] 
        freq[c] -= 1
    end
    freq.values.all? { |count| count == 0 }
end